pkg/collector/osmlogs_collector.go

```diff
@@ -41,96 +40,96 @@ func (collector *OsmLogsCollector) Collect() error {
-	}
-
-	for fileName, kubeCmds := range groundTruthMap {
-		if err = collector.CollectKubectlOutputToCollectorFiles(rootPath, fileName, kubeCmds); err != nil {
-			log.Printf("Failed to collect %s for OSM: %+v", fileName, err)
-		}
-	}
-
-	controllerNamespaces, err := utils.GetResourceList([]string{"get", "deployments", "--all-namespaces", "-l", "app=osm-controller", "-o", "jsonpath={..metadata.namespace}"}, " ")
-	if err != nil {
-		return err
-	}
-
 	for _, meshName := range meshList {
+		meshRootPath := filepath.Join(rootPath, "mesh_"+meshName)
 		monitoredNamespaces, err := utils.GetResourceList([]string{"get", "namespaces", "--all-namespaces", "-l", "openservicemesh.io/monitored-by=" + meshName, "-o", "jsonpath={..name}"}, " ")
 		if err != nil {
 			log.Printf("Failed to get namespaces within osm mesh '%s': %+v\n", meshName, err)
 			continue
 		}
-		if err = collectDataFromNamespaces(collector, monitoredNamespaces, controllerNamespaces, rootPath, meshName); err != nil {
+		controllerNamespaces, err := utils.GetResourceList([]string{"get", "deployments", "--all-namespaces", "-l", "app=osm-controller,meshName=" + meshName, "-o", "jsonpath={..metadata.namespace}"}, " ")
+		if err != nil {
+			return err
+		}
+		if err = callNamespaceCollectors(collector, monitoredNamespaces, controllerNamespaces, meshRootPath, meshName); err != nil {
 			log.Printf("Failed to collect data from OSM monitored namespaces: %+v", err)
 		}
-		if err = collectControllerLogs(collector, rootPath, meshName); err != nil {
-			log.Printf("Failed to collect OSM controller logs for mesh %s: %+v", meshName, err)
+		if err = collectGroundTruth(collector, meshRootPath, meshName); err != nil {
+			log.Printf("Failed to collect ground truth for mesh %s: %+v", meshName, err)
 		}
 	}
-
 	return nil
 }
 
-// ** Collects data for namespaces monitored by a given mesh and osm-controller namespace **
-func collectDataFromNamespaces(collector *OsmLogsCollector, monitoredNamespaces, controllerNamespaces []string, rootPath, meshName string) error {
+// Calls functions to collect data for osm-controller namespace and namespaces monitored by a given mesh
+func callNamespaceCollectors(collector *OsmLogsCollector, monitoredNamespaces, controllerNamespaces []string, rootPath, meshName string) error {
 	for _, namespace := range monitoredNamespaces {
-		if err := collectEnvoyData(collector, rootPath, meshName, namespace); err != nil {
+		namespaceRootPath := filepath.Join(rootPath, "namespace_"+namespace)
+		if err := collectDataFromEnvoys(collector, namespaceRootPath, namespace); err != nil {
 			log.Printf("Failed to collect Envoy configs in OSM monitored namespace %s: %+v", namespace, err)
 		}
+		if err := collectNamespaceResources(collector, namespaceRootPath, namespace); err != nil {
+			log.Printf("Failed to collect resource data for OSM monitored namespace %s: %+v", namespace, err)
```

Comment:
Can we do `log.Print("Failed to collect resource data)` since `collectNamespaceResources()` will capture the specific log error in L 90/L109?

Reply:
Removing this per our conversation offline because no error is ever returned here